### DISCUSSION TITLE
Online and Servers Trouble

### DISCUSSION THREAD

**Discussion by Kerrpllardy** [avaraline/Avara#177] [Q&A] [answered]
I was with some of my friends and we were trying to make a server to play on in the game. Our servers kept showing up in the "Tracker", but when any on my friends clicked "Connect..." nothing would happen. We were all in the same room together on the same wifi. I was using Mac and my two other friends were using Windows.

I tried messing with what I assumed was the player count number, but we had no luck. 
Has this feature not been fully finished? Are we doing something wrong or out of order?

The three of us love this game and love. how easy it was to just download and play, but we just couldn't get the server and tracker to work for us. 

Thank you in advance for any help.

**Comment by dcwatson** [comment, marked as answer]
There are still a lot of caveats with the networking in general. In your case (everyone on the same local network), you could have one person start hosting, then the other 2 would enter that person's *local* IP address (192.168.* or 10.* or 172.* or whatever) and press connect in the Network window (see below). The tracker only sees your public internet IP, so you can't use it for LAN games.

<img width="212" alt="Screen Shot 2021-06-26 at 3 19 44 PM" src="https://user-images.githubusercontent.com/293445/123523630-bf6b9480-d692-11eb-8cf6-0617bb0666eb.png">

If you do decide you want to play over the internet as well, you'll need to make sure you forward port 19567/udp on your router. We're working to make all of this easier in the future, but feel free to ask questions here or in the discord. Good luck!

**Reply by Kerrpllardy** [reply]
Thank you, we were able to get it to work!
Such a great project, and had we had a blast!

**Comment by rherriman** [comment]
Another thing I'd like to suggest you try is having two players modify their prefs.json file to connect via different ports. I believe this would allow you to use the tracker, and play together alongside players outside of your LAN group.

The prefs.json file locations are platform dependent, and outlined near the top of this page: https://github.com/avaraline/Avara/wiki/Configuring-your-Controls

There is a setting titled `udpDefaultPort` which ought to be set to 19567--I'd suggest that the other players change this value to 19568 and 19569. You would also need to do what @dcwatson indicated, however, and modify your router settings to forward those ports to the respective local IP addresses (e.g. 192.168.1.2, 192.168.1.3, or whatever they may be). Typically you'd want to ensure that those machines always connect to the router with the same IP, so you don't have to constantly change the port forwarding settings.

As @dcwatson also said, we know this is a huge pain point. It's been very upsetting for us to watch so many people try to play the game in the past week and being unable to figure it out. :) We hope we haven't driven them away forever. We're working on it! But the advice we've given you here _should_ get you through for the time being.

(In case you're curious about the reason why this is such a hassle right now: when the game was created, routers and WiFi in general were pretty much unheard of in everyday household use. There was no confusion about which IP address to use and port forwarding wasn't necessary because you were likely plugged directly into the modem. The code needs to be updated to reflect modern reality.)

**Comment by rherriman** [comment]
I wanted to point out that the current nightly build implements a change for all platforms that should greatly alleviate the need for manual port forwarding! Thank you @blenkush and @assertivist.
